### PR TITLE
- allow passing 0 as response

### PR DESCRIPTION
**Pull request by MrEssex**

### CODE REVIEW

**Comment by TomK** [comment]
i'm wondering if is_scalar to JsonResponse might be more appropriate, **except where is_string**
thoughts @bajb  ?

File: src/Routing/RouteProcessor.php [hunk 1]
```diff
@@ -131,6 +131,10 @@ protected function _processRoute(Context $c, $handler, &$response)
    */
   protected function _prepareResponse(Context $c, $result, $buffer = null)
   {
+    if(is_numeric($result))
+    {
+      return $result;
+    }
     return $result ?: ($buffer !== false ? $buffer : null);
```
Comment (bajb) [inline src/Routing/RouteProcessor.php:138]: maybe we just return $result instead of null here.

File: src/Routing/RouteProcessor.php [hunk 2]
```diff
@@ -98,7 +98,7 @@ protected function _processHandler(Context $c, $handler, &$response): bool
       return true;
     }
 
-    if($handler)
+    if($handler || is_numeric($handler))
```
Comment (bajb) [inline src/Routing/RouteProcessor.php:101]: whats the use case for this?

Comment (MrEssex) [reply src/Routing/RouteProcessor.php:101]: If the handler function just returns 0 then this wont be hit, not setting the response

**Review by TomK — APPROVED** [review]

**Pull request merged**